docs/modules/usage/how-to/custom-sandbox-guide.md

```diff
@@ -12,53 +12,131 @@ You have two options for customization:
 1. Use an existing image with the required software.
 2. Create your own custom Docker image.
 
-If you choose the first option, you can skip the `Create Your Docker Image` section.
+## Important Requirements
 
-## Create Your Docker Image
+Before proceeding with either option, note these key requirements:
+
+1. The base image must be Debian-based
+2. The user in the container MUST be `root` for proper functionality
+3. Any installed packages should be available system-wide, not just for specific users
 
-To create a custom Docker image, it must be Debian based.
+## Create Your Docker Image
 
-For example, if you want OpenHands to have `ruby` installed, create a `Dockerfile` with the following content:
+Here's a complete example of a custom runtime Dockerfile:
 
 ```dockerfile
 FROM debian:latest
 
 # Install required packages
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y \
+    ruby \
+    python3 \
+    nodejs \
+    && rm -rf /var/lib/apt/lists/*
+
+# Ensure we're running as root
+USER root
 ```
 
-Save this file in a folder. Then, build your Docker image (e.g., named custom-image) by navigating to the folder in
-the terminal and running::
+Save this file as `docker/runtime.Dockerfile`. Then, build your Docker image using one of these methods:
+
+### Method 1: Direct Docker Build
```

Comment:
This may be a personal thing but the `Method 1` and `Method 2` do different things. One of them just builds the custom docker image and the other actually runs everything. I would just keep the `Method 1` and show the user how to build it.

Reply:
The other problem with the docker compose is it has OpenHands version which again is another place that needs to be continuously updated

Reply:
Good points.
1. Let's add a comment in the docker-compose to check the require variables that can be passed.
2. Let's add a command to the docker build (Method 1) that links to the explanation on how to run the command and specify how to pass in the newly built docker-image as the runtime image.

Reply:
> I would just keep the Method 1 and show the user how to build it.
Agree, just keep Method 1 and then include instructions on how to set the SANDBOX_RUNTIME_CONTAINER_IMAGE env var (maybe also the equivalent config.toml setting) when running openhands.